lib/pages/filter/model/filter.dart

```diff
@@ -60,6 +61,30 @@ class Filter {
     return list;
   }
 
+  List<FilterNode> findNodeByPath(List<String> path){
+    List<FilterNode> result = [root];
+    if(path == null){
+      return result;
+    }
+    for(int i = 0; i< path.length; i++){
+      FilterNode aux = _findNodeByPathHelper(path[i], result.last);
+      if(aux == null){
+        return result;
+      }
+      result.add(aux);
+    }
+    return result;
+  }
+
+  FilterNode _findNodeByPathHelper(String nodeName, FilterNode parent){
+    for(int i = 0; i < parent.children.length; i++){
+      if(parent.children[i].name.compareTo(nodeName) == 0){
+        return parent.children[i];
+      }
+    }
+    return null;
```

Comment:
This can be a one-liner, not sure it needs to be a function.
```suggestion
  return parent.children.firstWhere((child) => child.name.equals(nodeName), null);
```